### PR TITLE
[DOC] Add Sktree sphinx documentation for unsupervised criterions

### LINKED ISSUE #15: Build Sphinx documentation locally and add section for FastBIC
**Is your feature request related to a problem? Please describe.**
Run sphinx documentation locally. Start a section for FastBIC

**Describe the solution you'd like**
First build sphinx via `sphinx-build -b html docs/ docs/build/html`. Then start adding contents

**Estimated completion date**
Rough timeframe of the project is 2/20/2023 - 2/26/2023. This should be completed relatively soon after the local doc is built

### PR DESCRIPTION
**Pull request by jshinm**
Fixes #15

Changes proposed in this pull request:

Extends criterion section of the scikit-learn tree documentation and adds two-means and fastBIC criterions for cluster section. 

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Comment by jshinm** [comment]
Fixed some issues with sphinx build, still has a few errors from missing files on `tree.rst`. Will finalize the content of the criterion documentation soon.

**Comment by adam2392** [comment]
Hard to review cuz mostly copy and paste. 

Can you remove sections that are not relevant to scikit-tree and instead just add a link to the relevant sklearn docs? E.g. we do not implement a decisiontreeclassifier.

**Comment by jshinm** [comment]
Thank you! When I push my update, I'll make sure irrelevant parts are excluded.

**Comment by jshinm** [comment]
@adam2392 this PR primarily focuses on unsupervised DT section and two criteria we've added. I still need to add some class references to get the link working, and I'll do that for this week's documentation PR (just so I won't have to deal with merge conflict). Please let me know how you feel about this doc at your convenience!

**Comment by adam2392** [comment]
This module is not rendered anywhere when we build the documentation. Can you create a user-guide page that is added to the main landing page, or link this in some way? 

I'll review it when it's rendered via sphinx.

See scikit-learn, or hyppo docs code for examples. Lmk if you have any questions.

**Comment by jshinm** [comment]
@adam2392 I've updated the doc in general (simplifying, restyling, rearranging sections). The tree doc update is under User Guide

**Comment by adam2392** [comment]
### Fix the CI, so it runs:

Not sure why the ssh-agent is not working. @jshinm can you try commenting out that section in the CI? 

If that does not work, you can change the requirements line for scikit-learn to not use ssh: `scikit-learn@git+https://github.com/neurodata/scikit-learn@tree-featuresv2`

**Comment by adam2392** [comment]
Try pushing again because the circleCI was default not building for forked pull requests. circleCI builds the docs, which is the primary focus of this PR.

**Comment by adam2392** [comment]
FYI you can run doc builds locally using the `dev.py` tool

**Comment by jshinm** [comment]
@adam2392 it seems to build fine locally, but it can't pass CI test :(

I tested with main branch to see if something's wrong, but it builds fine locally and passes CI, so I'm a bit confused, but I feel good about my last commit :crossed_fingers:

**Comment by adam2392** [comment]
It would be helpful for you to also fix the installation and other pages while we're on this:

- https://output.circle-artifacts.com/output/job/28564935-5ef2-4158-b704-0f4246143cf5/artifacts/0/html/modules/get_started/install.html making sure the documentation is correct (e.g. scikit-learn version should specify the fork)
- In the front page (https://output.circle-artifacts.com/output/job/28564935-5ef2-4158-b704-0f4246143cf5/artifacts/0/html/index.html), it should link to the "more in-depth" installation page as well
- Also because it copies the README, there's a bunch of stuff that does not make sense as a home-page.  E.g. "See here for the documentation for our dev version: https://docs.neurodata.io/scikit-tree/dev/index.html" makes sense in a README, but not in the front landing page.
- Any other discrepancies you can find by going through each page that were the result of copying some of the source code from scikit-learn we should aim to fix. E.g. one example is `New in version 0.17` at https://output.circle-artifacts.com/output/job/28564935-5ef2-4158-b704-0f4246143cf5/artifacts/0/html/generated/sktree.tree.UnsupervisedDecisionTree.html#sktree.tree.UnsupervisedDecisionTree

File: docs/modules/user_guide/tree.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
+.. _tree:
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:1]: I would name this module `unsupervised_tree` and also rename the file

File: docs/modules/user_guide/tree.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+===========================
+
+.. currentmodule:: sklearn.tree
+
+In unsupervised learning, the goal is to identify patterns 
+or structure in data without using labeled examples. Clustering is a common 
+unsupervised learning technique that groups similar examples together 
+based on their features. In clustering settings, there may be no natural 
+notion of “true” class-labels, thus the efficacy of the clustering scheme is 
+often measured based on distance based metrics such as :func:`sklearn.metrics.adjusted_rand_score`.
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:14]: Can you add a refernece/link to the scikit-learn documentation regarding clustering metrics? 

https://scikit-learn.org/stable/modules/clustering.html#clustering-performance-evaluation

File: docs/modules/user_guide/tree.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+One way to combine clustering with decision trees is to use a clustering 
+algorithm to partition the data into groups, and then build a decision tree 
+on each group. This approach is known as clustering-based decision trees or 
+clustering trees. Each decision tree only needs to consider a subset of 
+the data, which can lead to faster and more accurate predictions. Two of 
+the most common apporaches used for unsupervised classification are 
+K-Means clustering and Mixture Modeling, which use the Expectation 
+Maximization Algorithm.
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:23]: I don't understand this section. Can you elaborate?

File: docs/modules/user_guide/tree.rst [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+In clustering settings, there may be no natural notion of “true”
+class-labels; the efficacy of the clustering scheme is often measured 
+in such cases by the economy in description length attained
+by a two-step description of the objects by first describing the 
+attributes common to the clusters and then describing the differential
+attributes of each object within the cluster. k-Means Clustering
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:30]: I have a hard time understanding/reading phrases like "the economy in description length" and "describing the differential attributes of each object within the cluster". Is this referring to metrics? something else?

File: docs/modules/user_guide/tree.rst [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+and Mixture Modeling using the Expectation Maximization Algorithm 
+are examples of techniques used for unsupervised classification.
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:32]: What does EM and KMeans have to do with clustering with trees? I'm not following most of this section.

File: docs/modules/user_guide/tree.rst [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+Another way to use unsupervised learning with decision trees is to use 
+clustering to pre-process the data before training a supervised 
+decision tree. This can help identify patterns in the data and reduce 
+noise, which can improve the performance of the decision tree.
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:37]: Do we do that here?

File: docs/modules/user_guide/tree.rst [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+The resulting tree structure can be used to construct a decision tree by 
+selecting a threshold to cut the tree at a particular level.
+
+The advantage of using clustering criteria in decision trees is that 
+it can help identify subgroups of data with different characteristics, 
+which can lead to more accurate and interpretable models. However, the 
+choice of clustering criterion and the number of clusters can significantly 
+affect the performance of the decision tree. Therefore, it is important to 
+carefully evaluate different clustering criteria and parameter settings to 
+find the best approach for a particular dataset.
```
Comment (adam2392) [inline docs/modules/user_guide/tree.rst:74]: This entire section reads as if tho the TwoMeans and FastBIC objective is used to perform clustering with Means and Agglomerative Clustering.

**Pull request closed without merging**